aws-blog-event-driven-batch-analytics/src/main/java/com/amazonaws/bigdatablog/edba/emr/ProcessVendorTrasactions.java

```diff
@@ -0,0 +1,87 @@
+package com.amazonaws.bigdatablog.edba.emr;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.sql.DataFrame;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+import com.amazonaws.auth.AWSSessionCredentials;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.google.common.collect.ImmutableMap;
+
+import scala.collection.Seq;
+
+public class ProcessVendorTrasactions {
+
+	final static String redshiftJDBCURL="jdbc:redshift://<<YourClusterEndPoint>>:5439/test?user=<<YourUser>>&password=<<YourPassword>>";
+	public static void main(String[] args){
+		
+		    	SparkConf conf = new SparkConf().setAppName("Spark Redshift No Access-Keys");
+		    	List<StructField> schemaFields = new ArrayList<StructField>();
+		    	schemaFields.add(DataTypes.createStructField("vendor_id", DataTypes.StringType, true));
+		    	schemaFields.add(DataTypes.createStructField("trans_amount", DataTypes.StringType, true));
+		    	schemaFields.add(DataTypes.createStructField("trans_type", DataTypes.StringType, true));
+		    	schemaFields.add(DataTypes.createStructField("item_id", DataTypes.StringType, true));
+		    	schemaFields.add(DataTypes.createStructField("trans_date", DataTypes.StringType, true));
+		    	StructType schema = DataTypes.createStructType(schemaFields);
```

Comment:
I'd suggest that lots of these items can be setup on the constructor and cached for the lifetime of the object, no?